internal/configprovider/config_source_provider.go

```diff
@@ -109,6 +118,12 @@ func (c *configSourceConfigMapProvider) Get(context.Context) (*config.Map, error
 		return nil, err
 	}
 
+	// If we've already initiated and started a server, shut it down and create a new one for
+	// updated config sources.
+	if c.configServer != nil {
```

Comment:
Do you think it would be possible and desirable to move the server start (and stop) code out of this function? It seems like starting a server is an unfortunate side effect of this Get function.

Reply:
I originally wanted to be able to do that, but there isn't any clean and simple way to make it work. The config server is a Splunk distribution-specific thing, and the only call into the Splunk distro is a `Retrieve` call made on a single config location. The Splunk distro doesn't have the high level view of how many config sources there are, or what the configs are. These methods know if a config has been seen one before, but it don't know if the current one is the last, which is really what is needed to know to make it a simple "start once" operation.

The clean option would be to have a single call to start the server after we've retrieved all the config sources. However, this would require a change in the core repository, but that doesn't work since there's no concept of the config server there.